Add  support for configurable OTLP export strategy with cumulative default strategy

huyan0 opened the pull request:
This PR describes and records discussions around spec [#731](https://github.com/open-telemetry/opentelemetry-specification/issues/731), [#725](https://github.com/open-telemetry/opentelemetry-specification/issues/725) and the overall conclusion from Metrics SIG meetings. 

cc: @alolita @huyan0

Commit:
add otlp export otep

CunjunWang reviewed (commented):
LGTM

jmacd commented on text/metrics/0000-otlp-export-behavior.md:
```diff
@@ -0,0 +1,49 @@
+
+To support Prometheus as well as delta backends, the SDK should be configurable and supports OTLP exporter to have both cumulative default or delta export behavior. The implication is that the OTLP metric protocol should support both cumulative and delta report strategy.
+
+Users should be allowed to declare an environment variable or configuration field that determines this setting for OTLP exporters.
```
Maybe suggest a value?

jmacd reviewed (approved):
This is :100: great!

huyan0 commented on text/metrics/0000-otlp-export-behavior.md:
```diff
@@ -0,0 +1,49 @@
+
+ To support cumulative export strategy, the SDK needs to maintain state of each cumulative metrics. This means users with high-cardinality metrics can experience high memory usage. 
+ 
+This problem could be addressed by adding support in the Collector, when configured as an Agent, to support converting delat OTLP to Cumulative OTLP, but this requires a single agent for each metric generating client so that all delta values of a metric are converted at the same location. 
```
typo: delta

dabangarang commented on text/metrics/0000-otlp-export-behavior.md:
```diff
@@ -0,0 +1,49 @@
+
+## Motivation 
+
+1. Metric backends such as Prometheus and Prometheus Remote Write integrated storages requires cumulative values for cumulative/adding metrics rather than delta value per collection interval. Thus, to export metrics generated by the SDK using the collector, incoming values from the SDK should be in cumulative form. However, backends like Statsd expects delta value of each collection interval. To  ...
```
Maybe say "backends that make use of the Prometheus Remote Write API"

alolita replied:
# OTLP Configurable Export Behavior
 Add support for configurable behavior in OTLP exporters

Commit:
Make SDK OTLP exporters support configurable export behavior

Based on discussions from the Metrics SIG meeting, configurable export behavior from the OTLP exporter is desired to support different mertic backends

Commit:
Make SDK OTLP exporters support configurable export behavior

Based on discussions from the Metrics SIG meeting, configurable export behavior from the OTLP exporter is desired to support different mertic backends

jkwatson committed:
A proposal for SDK configurations for metric aggregation (Basic Views) (#126)

* Add a proposal for SDK configurations for metric aggregation.

* rename file to match the PR #

* fix markdown lint issues

* clarify that this applies to the default SDK, and fill out the open questions

* update the java example to fix the naming changes

* Add another open question to the list.

* Update text/0126-Configurable-Metric-Aggregations.md

Co-authored-by: Chris Kleinknecht <libc@google.com>

* Update to use the 'view' terminology

* another configuration/view replacement

* Add a few more open questions, and a note that they will be resolved in the spec.

* Update text/0126-Configurable-Metric-Aggregations.md

Co-authored-by: Tyler Yahn <MrAlias@users.noreply.github.com>

Co-authored-by: Chris Kleinknecht <libc@google.com>
Co-authored-by: Tyler Yahn <MrAlias@users.noreply.github.com>
Co-authored-by: Bogdan Drutu <bogdandrutu@gmail.com>

bogdandrutu committed:
Rename text/0126-Configurable-Metric-Aggregations.md to text/metrics/0126-Configurable-Metric-Aggregations.md (#132)

Commit:
fix format to pass markdownlint

Commit:
add sentence to highlight cumulative -> delta is also needed

bogdandrutu committed:
Merge branch 'master' into master
